### ISSUE TITLE
Wrong results for TPCH benchmarks when using native storage format

### ISSUE THREAD

**Issue by djouallah** [GlareDB/glaredb#1834]
Glaredb seems to not to return results at all for some queries when using Native format

https://colab.research.google.com/drive/11Jz3G1oqf8glye5BmNIq5cOdeUQJjImv#scrollTo=MDTt42NXEL5_

**Comment by universalmind303** [comment]
looks like something weird is going on with date math

this return an empty result set
```sql
select l_shipdate from lineitem where l_shipdate > date '1995-03-09';
```

but an explicit cast seems to work
```sql
select l_shipdate from lineitem where l_shipdate::timestamp > date '1995-03-09';
```

trying to figure out why the date math isn't working for `Date32` dtype.

**Comment by djouallah** [comment]
btw, it does works fine with parquet, I suspect the issue is happening when I convert parquet to delta 

https://colab.research.google.com/drive/1_iD6S6MR88B1Ym91pXksmtqaQzF4seQa#scrollTo=S2iF1xWKUXF7

**Comment by universalmind303** [comment]
@djouallah it appears to be an issue with the delta-rs library. We'll work on getting it fixed upstream! 

please refer to https://github.com/delta-io/delta-rs/issues/1670 if you want to track the progress.

**Comment by scsmithr** [comment]
@djouallah this has been fixed and is included in the latest 0.5.1 release.
